pkg/apis/apps.kubermatic/v1/application_definition.go

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	semverlib "github.com/Masterminds/semver/v3"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	HelmTemplateMethod TemplateMethod = "helm"
+)
+
+const (
+	GIT_AUTH_PASSWORD GitAuthMethod = "password"
+	GIT_AUTH_TOKEN    GitAuthMethod = "token"
+	GIT_AUTH_SSH_KEY  GitAuthMethod = "ssh-key"
+)
+
+const (
+	FORMFILED_TYPE_NUMBER    = "number"
+	FORMFILED_TYPE_BOOLEAN   = "boolean"
+	FORMFILED_TYPE_TEXT      = "text"
+	FORMFILED_TYPE_TEXT_AREA = "text-area"
```

Comment:
```suggestion
	FormFieldTypeNumber   FormFieldType = "number"
	FormFieldTypeBoolean  FormFieldType = "boolean"
	FormFieldTypeText     FormFieldType = "text"
	FormFieldTypeTextArea FormFieldType = "text-area"
```

Reply:
BTW, I'd recommend to move constants definition close to the type where kubebuilder:validation:Enum is written.

Reply:
thanks for the best practices :)